src/plugins/console/public/lib/mappings/mappings.js

```diff
@@ -330,74 +341,64 @@ export function retrieveAutoCompleteInfo(settings, settingsToRetrieve) {
     componentTemplates: settingsToRetrieve.templates,
   };
 
-  const mappingPromise = retrieveSettings('fields', settingsToRetrieve);
-  const aliasesPromise = retrieveSettings('indices', settingsToRetrieve);
-  const legacyTemplatesPromise = retrieveSettings('legacyTemplates', templatesSettingToRetrieve);
-  const indexTemplatesPromise = retrieveSettings('indexTemplates', templatesSettingToRetrieve);
-  const componentTemplatesPromise = retrieveSettings(
-    'componentTemplates',
-    templatesSettingToRetrieve
-  );
-  const dataStreamsPromise = retrieveSettings('dataStreams', settingsToRetrieve);
-
-  $.when(
-    mappingPromise,
-    aliasesPromise,
-    legacyTemplatesPromise,
-    indexTemplatesPromise,
-    componentTemplatesPromise,
-    dataStreamsPromise
-  ).done((mappings, aliases, legacyTemplates, indexTemplates, componentTemplates, dataStreams) => {
-    let mappingsResponse;
-    try {
-      if (mappings && mappings.length) {
-        const maxMappingSize = mappings[0].length > 10 * 1024 * 1024;
-        if (maxMappingSize) {
-          console.warn(
-            `Mapping size is larger than 10MB (${mappings[0].length / 1024 / 1024} MB). Ignoring...`
-          );
-          mappingsResponse = '[{}]';
-        } else {
-          mappingsResponse = mappings[0];
+  const promises = Object.keys(settingKeyToPathMap).map((settingKey) => {
```

Comment:
I see you tried to refactor that code. My recommendation here make that code more maintainable and use the previous approach with some modification.

I would like to see little unified methods like `retrieveDataStreams` or `retrieveAliases`

Something like: 

```
  const retrieveDataStreams = async (settingsToRetrieve) => {
    const dataStreams = await retrieveSettings('dataStreams', settingsToRetrieve);

    if (dataStreams) {
      loadDataStreams(dataStreams);
    }
  };

  Promise.allSettled([retrieveDataStreams()]).then(() => {
    // Schedule next request.
    pollTimeoutId = setTimeout(() => {
      // This looks strange/inefficient, but it ensures correct behavior because we don't want to send
      // a scheduled request if the user turns off polling.
      if (settings.getPolling()) {
        retrieveAutoCompleteInfo(http, settings, settings.getAutocomplete());
      }
    }, settings.getPollInterval());
  })
```

And please use more TS features to make our code a little bit more stable